### DISCUSSION TITLE
Gradle plugins instead of just full on project?

### DISCUSSION THREAD

**Discussion by mlasevich** [docToolchain/docToolchain#554] [Ideas]
Copying a full on repo is nice if you are starting from scratch, but is a pain when trying to figure out how to use a part of the project only. In my case I already have a full on gradle build that performs all the steps I need, including custom handling of adoc's, except I want to publish to Confluence. There seems to be no documentation on how to do this. I have to reverse engineer the repo to get what I need from it.

It would be nice and a lot more flexible  if this was a gradle plugin with a set of tasks people can include in their gradle build file and use as needed.

Thanks

**Comment by mlasevich** [comment]
*sigh* https://github.com/docToolchain/gradle-doctoolchain-plugin

**Reply by rdmueller** [reply]
this plugin is a start. Feel free to extend it.
you can also install docToolchain as command line tool (see the docs) but we currently also work on a wrapper. See this talk for more details: https://fosdem.org/2021/schedule/event/ttddoctoolchain/

The wrapper is already in a state where I would use it, but it hasn't been fully tested yet.

**Reply by mlasevich** [reply]
I could not get the plugin I linked to to work - many issues, but the instructions are off and the artifact does not match what is in git (different plugin name, for one) - even compiling from scratch did not work. :-( Can a working(even if beta) version be pushed to Gradle plugin repository?

**Reply by rdmueller** [reply]
as far as I know, the plugin is only a sketch, an idea and nobody worked on it further. Maybe, I will find some time after the wrapper is finished, but this will not be soon.
